nova/virt/vmwareapi/vm_util.py

```diff
@@ -1380,15 +1306,32 @@ def _get_vm_ref_from_extraconfig(session, instance_uuid):
                                      _get_object_for_optionvalue)
 
 
-@vm_ref_cache_from_instance
+class VmMoRefProxy(StableMoRefProxy):
+    def __init__(self, ref, uuid):
+        super(VmMoRefProxy, self).__init__(ref)
+        self._uuid = uuid
+
+    def fetch_moref(self, session):
+        vm_value_cache_delete(self._uuid)
+        self.moref = search_vm_ref_by_identifier(session, self._uuid)
+        if not self.moref:
+            raise exception.InstanceNotFound(instance_id=self._uuid)
+        vm_ref_cache_update(self._uuid, self.moref)
+
+    def __eq__(self, other):
+        try:
+            return self.moref == other.moref
```

Comment:
I thought this doesn't work for morefs as they are individual objects and don't compare on `.value`. I think we need `vutil.get_moref_value()` around both.

Reply:
It is only needed for the tests. I'll remove it from the object itself and move it to the tests in another patch.